File: case_study_furama/angular-furama/src/app/service-management/model/service.ts
import {RentType} from './rent-type';
import {ServiceType} from './service-type';

export interface Service {
  id?: number;
  area?: string;
  floor?: string;
  maxPeople?: string;
  name?: string;
  otherConvenience?: string;
  poolArea?: string;
  price?: string;
  serviceCode?: string;
  standardRoom?: string;
  rentType?: RentType;
  serviceType?: ServiceType;
  image?: string;
}

File: case_study_furama/spring-furama-sever/src/main/java/furama_sever/model/service/Service.java
package furama_sever.model.service;

import com.fasterxml.jackson.annotation.JsonBackReference;
import furama_sever.model.contract.Contract;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Getter
@Setter
@Entity
public class Service {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String area;
    private String floor;
    private String maxPeople;
    private String name;
    private String otherConvenience;
    private String poolArea;
    private String price;
    private String serviceCode;
    private String standardRoom;

    @ManyToOne
    @JoinColumn(name = "rent_type_id", referencedColumnName = "id")
    private RentType rentType;

    @ManyToOne
    @JoinColumn(name = "service_type_id", referencedColumnName = "id")
    private ServiceType serviceType;

    @JsonBackReference
    @OneToMany(mappedBy = "service")
    private Set<Contract> contracts;

    public Service() {
    }

}
